packages/trader/src/Stores/Modules/Trading/trade-store.ts

```diff
@@ -685,12 +685,15 @@ export default class TradeStore extends BaseStore {
     };
 
     async loadActiveSymbols(should_set_default_symbol = true, should_show_loading = true) {
+        if (this.is_dtrader_v2_enabled) {
+            await when(() => this.has_symbols_for_v2);
```

Comment:
Just for my understanding: it will run callback (predicate?) until this.has_symbols_for_v2 === true?

Reply:
waits until the condition is true then return executes